### ISSUE TITLE
Complating half selection

### ISSUE THREAD

**Issue by mehmetilker** [recogito/recogito-js#13]
Is is possible to extend selection start and end offset when user just mark middle of the word or few words.
Some thing like:

"This is example sentence cont[ains an anno]tation"
User select just "ains an anno" part but after onclick fired? selection will extend to "contains an annotation"

**Comment by rsimon** [comment]
The issue I see is that people will sometime want to select only parts of a word. Therefore, as a general feature I don't think this would be good. Possibly it could be some kind of optional setting.

Alternatively, I was thinking about functionality for editing existing selections: grab the left or right edge of a selection, and then expand or shrink it with the mouse. That would be a significant extension, however (= a lot of coding work).

In other words: possible, but probably not going to happen anytime soon since the main focus of my work is currently on the image annotation projects [Annotorious](https://github.com/recogito/annotorious) and [AnnotoriousOSD](https://github.com/recogito/annotorious-openseadragon).

**Comment by mehmetilker** [comment]
Well it would be nice even as an optional setting.
Or I can implement myself with some guidence if possible.
For example if I play with dom in this event would it break something or I can get annotations as it should be?

      // Add an event handler  
      r.on('createAnnotation', function(annotation) { /** **/ });

**Comment by efi** [comment]
But is it not easier to rely on the built-in capability of the browser that distinguishes both use cases already?

If you double click on a word you select it whole. If you drag your mouse before releasing the second mousedown your selection will cover whole words.

Many people are already aware of this mechanism and are accustomed to it.

**Comment by mehmetilker** [comment]
Problem is not a word selection but a phrase. Selection a phrase properly. Like NER annotation tools out there but in a simple way with tagging feature like this library provides..
Yes it is selecting two or tree words is also simple. But simple for me. From user perspective, handling problems of boring and repetitive task would be nice. Again with tagging feature like in this library. Instead of creating something new, just adding expending selection looks easier to me. After all what open source is all about?

**Comment by rsimon** [comment]
I agree automation for something like this would be nice for many use cases, where people want to achieve a specific task fast. But, as I said, not necessarily as the default way the selection should work.

Using the handler would be possible. A bit hacky & not entirely trivial, but feasible. The overall plan would be:

- Use `on('createAnnotation')` to listen to a new annotation getting created
- The annotation will contain the character offsets of the selected text section ("ains an anno"). Use the offsets to search the DOM text element and figure out the offsets for the nearest whitespaces to the left and right. That will give you the character offsets you want your "expanded" annotation to have.
- Clip out the expanded text snippet ("contains an annotation"), and create a __new__ annotation object
- Delete the annotation you got with the `createAnnotation` event, and add your modified annotation via `.addAnnotation`

**Comment by mehmetilker** [comment]
@rsimon  Thanks for guidance,  looks clear to me, I will try.
